internal/gw/s3_upload_client.go

```
@@ -0,0 +1,86 @@
+	if _, err := io.Copy(h, rs); err != nil {
+		return "", fmt.Errorf("computing Content-MD5: %w", err)
+	}
+
+	if _, err := rs.Seek(0, io.SeekStart); err != nil {
+		return "", fmt.Errorf("rewinding upload body after MD5: %w", err)
+	}
+
+	return base64.StdEncoding.EncodeToString(h.Sum(nil)), nil
+}
+
+func newS3Uploader(s3Client *s3.Client) *manager.Uploader {
```

golangci-lint (2.12.2)

[error] SA1019: manager.Uploader is deprecated: superceded by feature/s3/transfermanager. See https://github.com/aws/aws-sdk-go-v2/discussions/3306 

(staticcheck)